radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,330 @@
+#define PDB_RESPONSE_LENGTH			(2)
+
+/** Delay between telemetry read/write operations with the PDB on I2C is 5ms */
+#define PDB_I2C_TELEM_DELAY_MS		(5)
+
+/** Delay between other read/write operations with the PDB on I2C is 1ms */
+#define PDB_I2C_DELAY_MS			(1)
+
+/** Number of sun sensors on the CubeSat */
+#define NUM_SUN_SENSORS				(6)
+
+/** Number of other Telem Command Bytes */
+#define NUM_TELEM_CALLS 			(4)
+
+/** Filter Mask and Comparison value for Recognizing Telemetry Calls */
+#define TELEM_MASK 					(0xFF0000)
+#define TELEM_CALL_COMPARISON		(0x100000)
+
+
+/***************************************************************************************************
+                                          PRIVATE GLOBALS
+***************************************************************************************************/
+/**
+ *  Commands to request data for each of the 6 sun-sensors
+ *	located on each face of the CubeSat Solar Arrays.
+ */
+static uint32_t pdbSunSensorCommandBytes[6] = {
+		0x10E11C,     // SA1A - yPos
+		0x10E11D,     // SA1B - yNeg
+		0x10E12C,     // SA2A - xNeg
+		0x10E12D,     // SA2B - xPos
+		0x10E13C,     // SA3A - zNeg
+		0x10E13D      // SA3B - zPos
+};
+
+/**
+ * Commands for requesting output voltage readings for each bus and BCR
+ */
+static uint32_t pdbVoltageCommandBytes[4] = {
+		0x10E280,		// Output Voltage of BCR
+		0x10E220,		// Output Voltage of Battery Bus
+		0x10E210,		// Output Voltage of 5V Bus
+		0x10E200		// Output Voltage of 3.3V Bus
+};
+
+/**
+ * Commands for requesting output current readings for each bus and BCR
+ */
+static uint32_t pdbCurrentCommandBytes[4] = {
+		0x10E284,		// Output Current of BCR in mA
+		0x10E224,		// Output Current of Battery Bus
+		0x10E214,		// Output Current of 5V Bus
+		0x10E204		// Output Current of 3.3V Bus
+};
+
+/**
+ * Command for PDB Temperature Reading
+ */
+static uint32_t pdbTemperatureCommandBytes = 0x10E308;
+
+/**
+ * Command for Resetting PDB Watchdog (0x2200)
+ */
+static uint8_t pdbWatchdogResetCommand[2] = {0x22, 0x00};
+
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static int pdbTalk(uint8_t* command, uint8_t* response);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+/**
+ * Get the Irradiance (W/m^2) value from the sun sensor on each face of the CubeSat
+ *
+ * @param A pointer to a sun_sensor_status_t object, which will be filled with the converted ADC data for each sun sensor
+ * @return an error code, or 0 if no error
+ */
+int getSunSensorData(sun_sensor_status_t* sunData) {
```

Comment:
Function name should be prefixed with `pdb`; I recommend renaming to `pdbSunSensorData`